lib/Doctrine/Common/Cache/BashoRiakCache.php

```diff
@@ -0,0 +1,327 @@
+            // Do nothing
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFlush()
+    {
+        try {
+            $command = (new Command\Builder\ListObjects($this->riak))
+                ->buildBucket($this->bucketName)
+                ->acknowledgeRisk(true)
+                ->build();
+
+            $response = $command->execute();
+
+            if ($response->isNotFound()) {
+                return false;
+            }
+
+            $keys = $response->getKeys();
+
+            foreach ($keys as $location) {
+                try {
+                    $command = (new Command\Builder\DeleteObject($this->riak))
+                        ->atLocation($location)
+                        ->build();
+
+                    $command->execute();
+                } catch (RiakException $e) {
+                    // Do nothing
+                }
+            }
+
+            return true;
+        } catch (RiakException $e) {
+            // Do nothing
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doGetStats()
+    {
+        // Only exposed through HTTP stats API, not Protocol Buffers API
+        try {
+            $command = (new Command\Builder\FetchStats($this->riak))
+                ->build();
+
+            $response = $command->execute();
+
+            return $response->getAllStats();
+        } catch (RiakException $e) {
+            // Do nothing
+        }
+    }
+
+    /**
+     * Check if a given Riak Object have expired.
+     *
+     * @param \Basho\Riak\Object $object
+     *
+     * @return bool
+     */
+    private function isExpired(Object $object)
+    {
+        $metadataMap = $object->getMetadata();
+
+        return isset($metadataMap[self::EXPIRES_HEADER])
+            && $metadataMap[self::EXPIRES_HEADER] < time();
+    }
+
+    /**
+     * On-read conflict resolution. Applied approach here is last write wins.
+     * Specific needs may override this method to apply alternate conflict resolutions.
+     *
+     * {@internal Riak does not attempt to resolve a write conflict, and store
+     * it as sibling of conflicted one. By following this approach, it is up to
+     * the next read to resolve the conflict. When this happens, your fetched
+     * object will have a list of siblings (read as a list of objects).
+     * In our specific case, we do not care about the intermediate ones since
+     * they are all the same read from storage, and we do apply a last sibling
+     * (last write) wins logic.
+     * If by any means our resolution generates another conflict, it'll be up to
+     * the next read to properly solve it.}
+     *
+     * @param string   $id
+     * @param string   $vClock
+     * @param string   $expires
+     * @param Object[] $objectList
+     *
+     * @return \Basho\Riak\Object|null
+     */
+    protected function resolveConflict($id, $vClock, $expires, array $objectList)
```

Comment:
Please add the type declaration for arguments and return, we're on PHP 7.1 so you can use ` : ?Object`